### DISCUSSION TITLE
Why does Uniswap not have payable functions ?

### DISCUSSION THREAD

**Discussion by TilakMaddy** [Cyfrin/security-and-auditing-full-course-s23#50] [Q&A] [answered]
At least for the v1 where we know it's weth why not just have `payable` functions to directly pay eth to the contract? Isn't it more convenient than `transferFrom(..)` because that needs approvals and stuff. They could have avoided all that !

**Comment by PatrickAlphaC** [comment, marked as answer]
Great question! This is a common confusing part for engineers. 

`payable` functions only work for `eth` or _native gas token_ functions. 

For example:

```javascript
function doStuff() public payable {
  require(msg.value > 1); 
}
```

You actually can't `transferFrom` with ETH/Native Gas tokens!

ERC20s on the other hand, need to use functions like `transferFrom` since they are smart contracts themselves.

They don't need to be payable, since they don't use ETH!

```javascript
function doStuff() public {
  IERC20(some_address).transferFrom(address_a, address_b, amount);
}
```

Does that make sense?

**Reply by TilakMaddy** [reply]
Yes 👍

**Comment by KulzeinepHub** [comment]
Ok!

**Comment by Mixaakisev** [comment]
thx bro

**Comment by Vasilii098** [comment]
thx, gj

**Comment by Avkubs** [comment]
This is a quality solution
